lib/connection_controller.go

```diff
@@ -137,6 +138,26 @@ func (cc *ConnectionController) startValidatorConnector() {
 	}
 }
 
+// startNonValidatorConnector is responsible for ensuring that the node is connected to the target number of outbound
+// and inbound remote nodes. To do this, it periodically checks the number of outbound and inbound remote nodes, and
+// if the number is above the target number, it disconnects the excess remote nodes. If the number is below the target
+// number, it attempts to connect to new remote nodes.
+func (cc *ConnectionController) startNonValidatorConnector() {
+	cc.startGroup.Done()
+
+	for {
+		select {
+		case <-cc.exitChan:
```

Comment:
Same comment as in the ValidatorConnector PR. This thread won't be able to read from the existChan. Please make sure this is safe